src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/FindOrganisationsByPbaStatusTest.java

```diff
@@ -0,0 +1,152 @@
+import uk.gov.hmcts.reform.professionalapi.controller.response.OrganisationsWithPbaStatusResponse;
+import uk.gov.hmcts.reform.professionalapi.domain.PbaStatus;
+import uk.gov.hmcts.reform.professionalapi.util.AuthorizationEnabledIntegrationTest;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphabetic;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static uk.gov.hmcts.reform.professionalapi.helper.OrganisationFixtures.someMinimalOrganisationRequest;
+
+@SuppressWarnings("unchecked")
+public class FindOrganisationsByPbaStatusTest extends AuthorizationEnabledIntegrationTest {
+
+    @Before
+    public void setUpTestData() {
+        setUpTestDataForPba();
+    }
+
+    @Test
+    public void get_request_returns_organisations_with_accepted_pba_status() throws JsonProcessingException {
+        var orgPbaResponse = (List<OrganisationsWithPbaStatusResponse>)
+                professionalReferenceDataClient.findOrganisationsByPbaStatus(
+                        PbaStatus.ACCEPTED.toString(), hmctsAdmin, Boolean.FALSE);
+
+        validatePbaResponse(orgPbaResponse, 2, PbaStatus.ACCEPTED);
+    }
+
+    @Test
+    public void get_request_returns_organisations_with_pending_pba_status() throws JsonProcessingException {
+        var orgPbaResponse = (List<OrganisationsWithPbaStatusResponse>)
+                professionalReferenceDataClient.findOrganisationsByPbaStatus(
+                        PbaStatus.PENDING.toString(), hmctsAdmin, Boolean.FALSE);
+
+        validatePbaResponse(orgPbaResponse, 1, PbaStatus.PENDING);
+    }
+
+
+    @Test
+    public void get_request_returns_no_organisations_for_given_pba_status() throws JsonProcessingException {
+        var orgPbaResponse = (List<OrganisationsWithPbaStatusResponse>)
+                professionalReferenceDataClient.findOrganisationsByPbaStatus(
+                        PbaStatus.REJECTED.toString(), hmctsAdmin, Boolean.FALSE);
+
+        assertThat(orgPbaResponse).isEmpty();
+    }
+
+    @Test
+    public void get_request_returns_400_when_invalid_pba_status() throws JsonProcessingException {
+
+        Map<String, Object> errorResponseMap = (Map<String, Object>) professionalReferenceDataClient
+                .findOrganisationsByPbaStatus("Invalid Status", hmctsAdmin, Boolean.FALSE);
+        ErrorResponse errorResponse = (ErrorResponse) errorResponseMap.get("response_body");
+
+        assertThat(errorResponseMap).isNotNull().containsEntry("http_status", HttpStatus.BAD_REQUEST);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo(ErrorConstants.INVALID_REQUEST.getErrorMessage());
+    }
+
+    @Test
+    public void get_request_returns_400_when_null_pba_status() throws JsonProcessingException {
```

Comment:
Also, because pba status is passed as a path param, it should not allow comma separated values. You might want to include this scenario as well.

Reply:
thanks for the explanation @viswajit-kuna . I have modified the query, please review 6faeed5